review change (#8635)
pingcap/docs: develop/dev-guide-unstable-result-set.md

For convenience reasons, MySQL "extends" the `GROUP BY` syntax to allow the `SELECT` clause to refer to non-aggregated fields not declared in the `GROUP BY` clause, i.e., the `NON-FULL GROUP BY` syntax. In other databases, this is considered a syntax **_ERROR_** because it causes the result set to be unstable.

> For example, we have two tables: `stu_info` to save student information data and `stu_score` to holdsave student test scores.

Reviewer: ```suggestion
For example, you have two tables:

- `stu_info` stores the student information
- `stu_score` stores the student test scores.
```

{{< copyable "sql" >}}

```sql
SELECT
    `a`.`class`,
    `a`.`stuname`,
    max( `b`.`courscore` )
FROM
    `stu_info` `a`
    JOIN `stu_score` `b` ON `a`.`stuno` = `b`.`stuno`
GROUP BY
    `a`.`class`
ORDER BY
    `a`.`class`,
    `a`.`stuname`;
```

There will be two results that match this SQL:

- First result:

   ```sql
   +------------+--------------+------------------------+
   | class      | stuname      | max( `b`.`courscore` ) |
   +------------+--------------+------------------------+
   | 2018_CS_01 | MonkeyDLuffy |                   95.5 |
   | 2018_CS_03 | PatrickStar  |                   99.0 |
   +------------+--------------+------------------------+
   ```

- Second result:

   ```sql
   +------------+--------------+------------------+
   | class      | stuname      | max(b.courscore) |
   +------------+--------------+------------------+
   | 2018_CS_01 | MonkeyDLuffy |             95.5 |
   | 2018_CS_03 | SpongeBob    |             99.0 |
   +------------+--------------+------------------+
   ```

> This happens because you did **_NOT_** specify how to value the `a`.`stuname` field in the SQL, and both results are consistent with the SQL semantics. This results in an unstable result set.Therefore, if you want to guarantee the stability of `GROUP BY` statement result set, please use `FULL GROUP BY` syntax.

Reviewer: ```suggestion
It happens because you did **_NOT_** specify how to get the value of the `a`.`stuname` field in SQL, and two results are both satisfied by SQL semantics. This results in an unstable result set. Therefore, if you want to guarantee the stability of `GROUP BY` statement result set, use the `FULL GROUP BY` syntax.
```
